### PR TITLE
updated support for benchmark language versions

### PR DESCRIPTION
**Pull request by octonawish-akcodes**
Need reviews cc @mcopik 

Related issue #196 

## Summary by CodeRabbit

- **New Features**
  - Added `/alive` route to Node.js and Python servers that returns a status response.
  - Introduced a new `url` property to the `Function` class for better URL handling.

- **Dependency Updates**
  - Updated `sharp` dependency for Node.js `thumbnailer` module.
  - Added `pillow` package and specific versions for multiple benchmarks.
  - Introduced `igraph` package for scientific benchmarks.
  - Updated Python and Node.js versions in configuration files.

- **Enhancements**
  - Enhanced Azure function deployment with updated version settings.
  - Improved server setup in the local environment with a new waiting mechanism.

- **Documentation**
  - Updated benchmark documentation for GCP compatibility issues with specific Python versions.

- **Bug Fixes**
  - Refined function names and handlers in Python server for clarity.

### CODE REVIEW

File: config/systems.json [hunk 1]
```diff
@@ -46,7 +46,9 @@
           "files": [
             "storage.js"
           ],
-          "packages": []
+          "packages": {
+            "uuid": "3.4.0"
```
Comment (mcopik) [inline config/systems.json:50]: Why this change?

File: config/systems.json [hunk 2]
```diff
@@ -170,7 +170,6 @@
       },
       "nodejs": {
         "base_images": {
-          "10": "gcr.io/google-appengine/nodejs",
           "12": "gcr.io/google-appengine/nodejs",
```
Comment (mcopik) [inline config/systems.json:173]: What's the situation on Azure/GCP?

**Review by mcopik — CHANGES_REQUESTED** [review]

**Comment by mcopik** [comment]
@octonawish-akcodes Hi, any update on my comments?

File: config/systems.json (hunk 2 above)
Comment (octonawish-akcodes) [reply config/systems.json]: I checked the python, nodejs version supports on gcp, azure, openwhisk  official documentation most of them supports python versions 3.8 and above and for nodejs 18 and above.

**Comment by octonawish-akcodes** [comment]
@mcopik I updated the PR with the changes, have a look and happy to answer for further queries :)

**Comment by mcopik** [comment]
@octonawish-akcodes Thank you! On which platforms and languages did you verify it works? We need to run the benchmarks as some of them might break, e.g., due to a different language version required. For example, a common issue is the `Pillow` library for the thumbnail benchmark :) 

We can use SeBS regression to simplify the process, e.g., `./sebs.py benchmark regression test --config config/example.json --language nodejs --language-version 18 --deployment gcp`

**Comment by octonawish-akcodes** [comment]
Hi, I verfied them against the official documentation of the cloud platforms. Supported links are as follows:
For Azure:
Python (https://learn.microsoft.com/en-us/azure/azure-functions/functions-reference-python?tabs=asgi%2Capplication-level&pivots=python-mode-decorators#python-version)

Nodejs( https://azure.microsoft.com/en-in/updates/generally-available-azure-functions-support-for-nodejs-18/ )

For GCP:

Python (https://devguide.python.org/versions/#supported-versions)
Nodejs (https://github.com/googleapis/google-cloud-node)

For openwhisk

Python and Nodejs (https://openwhisk.apache.org/downloads.html)

**Comment by mcopik** [comment]
Yes, that is a very important first step. However, we need also to execute benchmarks against specified version to check they work in practice.

**Comment by octonawish-akcodes** [comment]
I can only try on the Azure platform since I don't have accounts on other platforms

**Comment by mcopik** [comment]
@octonawish-akcodes I think we can also add Python 3.10 on AWS Lambda.

**Comment by octonawish-akcodes** [comment]
> @octonawish-akcodes I think we can also add Python 3.10 on AWS Lambda.

I just checked it https://aws.amazon.com/about-aws/whats-new/2023/04/aws-lambda-python-3-10/ I'll add it

**Comment by octonawish-akcodes** [comment]
Can you share the failure, curious to see the traceback @prajinkhadka

**Comment by mcopik** [comment]
@prajinkhadka 411 is expected, we documented it :-) It just won't work without Docker images due to code size.

210 is likely a fixed Pillow version, and 50* is likely a fixed igraph version.

**Comment by prajinkhadka** [comment]
Most probably it is the issue with the versions as mentioned by Marcin. 

Regarding the 411 the issue should be with the code size limitation. I will check and update here if it's something else.

**Comment by octonawish-akcodes** [comment]
It will be better if u could post the whole traceback here.

**Comment by mcopik** [comment]
@prajinkhadka  Can you please post the correct versions here - or even better if both you and @octonawish-akcodes can push to the same branch.

**Comment by prajinkhadka** [comment]
@mcopik Yes, I will be doing it.

**Comment by octonawish-akcodes** [comment]
On Azure platform when i try to experiment benchmarks with python version 3.8, I am getting this error and even after exporting the certification using this command `export PYCURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt` it still gave me the same error, any comment @mcopik 

```
-------------

sebs.regression.AzureTestSequencePython.test_azure_210.thumbnailer: fail
sebs.regression.AzureTestSequencePython.test_azure_210.thumbnailer: Traceback (most recent call last):
  File "/home/abhi_ubuntu/sebs/serverless-benchmarks/sebs/regression.py", line 86, in test
    ret = trigger.sync_invoke(input_config)
  File "/home/abhi_ubuntu/sebs/serverless-benchmarks/sebs/azure/triggers.py", line 35, in sync_invoke
    return self._http_invoke(payload, self.url)
  File "/home/abhi_ubuntu/sebs/serverless-benchmarks/sebs/faas/function.py", line 206, in _http_invoke
    c.perform()
pycurl.error: (77, 'error setting certificate verify locations:\n  CAfile: /etc/pki/tls/certs/ca-bundle.crt\n  CApath: none')

```

**Comment by octonawish-akcodes** [comment]
Also on running the benchmarks against python version 3.10, all benchmarks are failed to execute.

```
-------------

sebs.regression.AzureTestSequencePython.test_azure_210.thumbnailer: fail
sebs.regression.AzureTestSequencePython.test_azure_210.thumbnailer: Traceback (most recent call last):
  File "/home/abhi_ubuntu/sebs/serverless-benchmarks/sebs/regression.py", line 66, in test
    func = deployment_client.get_function(
  File "/home/abhi_ubuntu/sebs/serverless-benchmarks/sebs/faas/system.py", line 147, in get_function
    raise Exception(
Exception: Unsupported python version 3.10 in azure!


-------------
```

**Comment by octonawish-akcodes** [comment]
On Azure with python version 3.8 test results:
image

cc @mcopik

**Comment by octonawish-akcodes** [comment]
Python 3.9 results: 
image

**Comment by mcopik** [comment]
@octonawish-akcodes It looks good. Thanks for the update!

What's the status of Python 3.10?

**Comment by octonawish-akcodes** [comment]
> @octonawish-akcodes It looks good. Thanks for the update!
> 
> What's the status of Python 3.10?

Platform error

**Comment by mcopik** [comment]
Update to PR:
- AWS support confirmed up to Python 3.11 and Node.js 16. For Python 3.12, we need to update Dockerfile to the new Amazon Linux (#208). For Node.js 18 and 20, we need to support AWS SDK v3 (#207)
- Azure now supports Python 3.7-3.10 with runtime 4.0, and Node.js 16-20.

**Review by mcopik — APPROVED** [review]

**Comment by mcopik** [comment]
@octonawish-akcodes @prajinkhadka Thank you for your help!

**Pull request merged**